packages/core/src/html/NodeProcessor.js

```diff
@@ -166,11 +181,13 @@ class NodeProcessor {
         this.mdAttributeRenderer.processTooltip(node);
         break;
       case 'modal':
-        // Transform deprecated slot names; remove when deprecating
-        renameSlot(node, 'modal-header', 'header');
-        renameSlot(node, 'modal-footer', 'footer');
+        if (!this._removeDuplicateModals(node)) {
```

Comment:
Should we explore extracting this logic (including line 186-189) into a process modal function instead? So
```js
    this._processModal(node)
   break;
```
and then something like:
```js
  _processModal(node) {
    if (this.processedModals[node.attribs.id]) {
       cheerio(node).remove();
    } else {
    this.processedModals[node.attribs.id] = true;
    // Transform deprecated slot names; remove when deprecating
          renameSlot(node, 'modal-header', 'header');
          renameSlot(node, 'modal-footer', 'footer');
          this.mdAttributeRenderer.processModalAttributes(node);
    }
  }
```

Reply:
Thanks @tlylt. Sounds good 👍